src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,127 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import LoadingText from 'amo/components/LoadingText';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import translate from 'amo/i18n/translate';
+import type { ResultShelfType } from 'amo/reducers/home';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type DefaultProps = {|
+  placeholderCount: number,
+|};
+
+type InternalProps = {|
+  i18n: I18nType,
+|};
+
+type Props = {|
+  ...DefaultProps,
+  ...InternalProps,
+  loading: boolean,
+  placeholderCount?: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  /* eslint-disable react/no-array-index-key */
+  static defaultProps: DefaultProps = {
+    placeholderCount: LANDING_PAGE_EXTENSION_COUNT,
+  };
+
+  homepageShelves: Array<React.Node> = this.props.shelves.map((shelf) => {
+    const { addons, criteria, endpoint, footer, title } = shelf;
+    const MOZILLA_USER_ID = config.get('mozillaUserId');
+    const titleStr = title.toString();
+    const header = titleStr.replace(/\s/g, '-');
+
+    const footerText =
+      footer && footer.text
+        ? footer.text
+        : this.props.i18n.sprintf(
+            this.props.i18n.gettext('See more %(categoryName)s'),
+            {
+              categoryName: titleStr.toLowerCase(),
+            },
+          );
+
+    const addonInstallSource =
+      endpoint === 'collections'
+        ? INSTALL_SOURCE_FEATURED_COLLECTION
+        : INSTALL_SOURCE_FEATURED;
+
+    const count =
+      endpoint === 'search-themes'
+        ? LANDING_PAGE_THEME_COUNT
+        : this.props.placeholderCount;
+
+    let footerLinkHtml;
+    if (footer && footer.url) {
+      footerLinkHtml = <a href={footer.url}>{footerText}</a>;
```

Comment:
This isn't right, we shouldn't have to put html here. I can see from `LandingAddonsCard` that it expects `footerLink` to be an internal URL, so if we want to support an external URL we'll need to change that as well. I have asked a question in the issue to see if we will need to support external URLs.

Reply:
See above, we can assume it will always be an internal URL and just pass it into the `to`.